plugins/alerting/pkg/alerting/api_conditions.go

```diff
@@ -343,28 +261,35 @@ func (p *Plugin) AlertConditionStatus(ctx context.Context, ref *corev1.Reference
 		return nil, shared.WithInternalServerError("cannot parse response body into expected api struct")
 	}
 	for _, alert := range respAlertGroup {
+		matched := false
```

Comment:
This logic is somewhat hard to follow. Maybe refactor such that the matched variable isn't needed

Reply:
Turns out this logic is correct iff len(router.HasLabels(conditionId)) == 1, so I'll rewrite & tidy up this portion